tests/test_models.py

```diff
@@ -106,4 +106,21 @@ def test_find_recommendation_type(self):
         self.assertEqual(recommendation.product_id2, recommendations[1].product_id2)
         self.assertEqual(recommendation.relationship, recommendations[1].relationship)
 
-    
\ No newline at end of file
+    def test_update_a_recommendation(self):
+        """Update a recommendation type by two product ids"""
+        recommendation = RecommendationFactory()
+        logging.debug(recommendation)
+        recommendation.create()
+        logging.debug(recommendation)
+        logging.debug(type(recommendation.relationship.name))
+        recommendation.relationship = Type.CROSS_SELL
+        recommendation.update()
+        self.assertIsNot(recommendation, None)
+        self.assertEqual(recommendation.relationship.name, 'CROSS_SELL')
+        recommendations = recommendation.all()
+        self.assertEqual(len(recommendations), 1)
+        self.assertEqual(recommendations[0].product_id1, recommendation.product_id1)
+        self.assertEqual(recommendations[0].product_id2, recommendation.product_id2)
+        self.assertEqual(recommendations[0].relationship, recommendation.relationship)
+
```

Comment:
To cover DataValidationError, you could create a new test and say here relationship = None, and then dataValidationError should be thrown.

Reply:
Done in test_model thanks